include/MidiFile.hpp

```diff
@@ -0,0 +1,317 @@
+#ifndef __MIDIFILE_HPP
+#define __MIDIFILE_HPP
```

Comment:
leading underscores in header guards is a no-no. Also, we don't use the .hpp extension, all our headers are .h